### DISCUSSION TITLE
Whoogle fails to start on Archlinux after sucessful installation with pipx - problem with libffi ? crypto ?

### DISCUSSION THREAD

**Discussion by m040601** [benbusby/whoogle-search#498] [General]
**Describe the bug**

Whoogle fails to start on Archlinux after successfull installation with pipx -problem with libffi ? crypto ?

**To Reproduce**

For some months now I've been testing whoogle search on Archlinux.
From all the installation methods, I've been using:

`pipx install --verbose whoogle-search`

It has worked without a problem in the last months. 
It stopped working a few days ago.
That is, it seems to install fine just like before.

But then after installation trying to start it:

    $ whoogle-search --help

```
Traceback (most recent call last):
  File "/home/a1/.local/bin/whoogle-search", line 5, in <module>
    from app.routes import run_app
  File "/home/a1/.local/pipx/venvs/whoogle-search/lib/python3.9/site-packages/app/__init__.py", line 1, in <module>
    from app.filter import clean_query
  File "/home/a1/.local/pipx/venvs/whoogle-search/lib/python3.9/site-packages/app/filter.py", line 5, in <module>
    from cryptography.fernet import Fernet
  File "/home/a1/.local/pipx/venvs/whoogle-search/lib/python3.9/site-packages/cryptography/fernet.py", line 18, in <module>
    from cryptography.hazmat.primitives import hashes, padding
  File "/home/a1/.local/pipx/venvs/whoogle-search/lib/python3.9/site-packages/cryptography/hazmat/primitives/padding.py", line 13, in <module>
    from cryptography.hazmat.bindings._padding import lib
ImportError: libffi.so.7: cannot open shared object file: No such file or directory
```

**Deployment Method**

- [ ] pip/pipx

**Version of Whoogle Search**
- [ ] Latest build from [source] (i.e. GitHub, Docker Hub, pip, etc)

**Desktop (please complete the following information):**

Archlinux 
Linux 5.14.14-arch1-1 #1 SMP PREEMPT Wed, 20 Oct 2021 21:35:18 +0000 x86_64 GNU/Linux

**Additional context**

I see also this error message "libffi" in this issue, [Manjaro update breaks Whoggle · Issue #476 · benbusby/whoogle-search](https://github.com/benbusby/whoogle-search/issues/476)

And since Archlinux and Manjaro are related, I strongly suspect it might be the same issue. But I'm not sure.

https://github.com/benbusby/whoogle-search/issues/476

**Comment by benbusby** [comment]
Seems like this is likely the same issue described in the other one you referenced. In any case, since this isn't an actual "bug" with Whoogle itself, I'm going to move this to a discussion instead and maybe others can chime in with what fixed it. 

Alternatively you could try installing the latest test pypi package of the app and see if that gives you the same error: https://test.pypi.org/project/whoogle-search/ -- if that works, I'll push a new version to regular pip. There were a couple libraries (unrelated to crypto) that were updated, so maybe there's some weird issue going on with that.

**Comment by harens** [comment]
Installing `libffi7`fixed this for me.

I couldn't find an apt package for my armv7l raspberry pi, so I downloaded it directly from [debian upstream](https://packages.debian.org/sid/libffi7).
